Update build

tomdee opened the pull request.

djosborne commented on Makefile:
```diff
@@ -1,30 +1,77 @@
+$(NODE_CONTAINER_BIN_DIR)/calico-felix:
+	-docker rm -f calico-felix
+	# TODO - update the felix image when https://github.com/projectcalico/calico/pull/1111 is merged
+	docker create --name calico-felix calico/felix:pyinstaller
```
```
# Latest felix binaries are stored in automated builds of calico/felix.
# To get them, we pull that image, then copy the binaries out to our host
```

djosborne commented on Makefile:
```diff
@@ -1,30 +1,77 @@
+	docker create --name calico-felix calico/felix:pyinstaller
+	docker cp calico-felix:/code/. $(@D)
+
+$(NODE_CONTAINER_BIN_DIR)/confd:
```
`# Get the confd binary`

djosborne commented on Makefile:
```diff
@@ -1,30 +1,77 @@
+	curl -L https://github.com/projectcalico/confd/releases/download/v0.10.0-scale/confd.static -o $@
+	chmod +x $@
+
+$(NODE_CONTAINER_BIN_DIR)/bird:
```
`# Get bird binaries`

djosborne commented on Makefile:
```diff
@@ -38,22 +85,41 @@ dist/calicoctl: $(CALICOCTL_FILE) birdcl gobgp
+	osrg/gobgp \
+	-c 'cp /go/bin/gobgp /code'
+
+simple-binary: $(CALICOCTL_FILE) birdcl gobgp
```
I thought `birdcl` and `gobgp` were `calico/node` deps, not `calicoctl` deps

tomdee replied:
You are incorrect, let's discuss offline.

tomdee commented on calico_node/Dockerfile:
```diff
@@ -14,29 +14,23 @@
+    apk add glibc-2.23-r3.apk glibc-bin-2.23-r3.apk && \
+    /usr/glibc-compat/sbin/ldconfig /lib /usr/glibc/usr/lib && \
+    apk del wget ca-certificates && \
+    rm -f glibc-2.23-r3.apk glibc-bin-2.23-r3.apk
```
An apk is like a deb. We download it, then install it. There's no need to leave the file lying around wasting space.

tomdee commented on calico_node/build.sh:
```diff
@@ -1,46 +0,0 @@
-pip install git+https://github.com/projectcalico/libcalico.git
-
-# Output the python library list
-pip list > libraries.txt
```
I has no meaning now. There is no pip and no pythong

tomdee replied:
*It

tomdee commented on Makefile:
```diff
@@ -1,30 +1,77 @@
+calico-node-latest.aci: calico-node.tar
+	docker2aci $<
+
+$(NODE_CONTAINER_CREATED): $(NODE_CONTAINER_DIR)/Dockerfile  $(addprefix $(NODE_CONTAINER_BIN_DIR)/,$(NODE_CONTAINER_BINARIES))
```
Yes, that's the whole point of a Makefile. If I do two builds in a row the second is instant. If I change a file, then make knows what to rebuild

djosborne commented on Makefile:
```diff
@@ -38,22 +93,41 @@ dist/calicoctl: $(CALICOCTL_FILE) birdcl gobgp
 	 pyinstaller calicoctl.spec -ayF
 
-simple-binary:
+birdcl:
```
`gobgp` and `birdcl` are dropped into project root on build. Can we volume mount pwd/dist instead?

djosborne replied:
```
# Get birdcl binary
```

djosborne commented:
I've got two untracked files after building that should get added to the gitignore
```
        allocate-ipip-addr.spec
        calico_node/filesystem/bin/
```

djosborne commented on Makefile:
```diff
@@ -38,22 +93,41 @@ dist/calicoctl: $(CALICOCTL_FILE) birdcl gobgp
+	wget -N https://github.com/projectcalico/calico-bird/releases/download/v0.1.0/birdcl
+	chmod +x birdcl
+
+gobgp:
```
Might as well add the last few generic comments:
```
# Get gobgp binary
```

djosborne commented:
Since some of these steps don't have a comment the line above, `make help` isn't very useful anymore. I could consider it out of scope, but maybe raise an issue to ensure all relevant build steps appear in  `make help` output?

djosborne commented on Makefile:
```diff
@@ -1,30 +1,85 @@
+# Get felix binaries
+$(NODE_CONTAINER_BIN_DIR)/calico-felix:
+	-docker rm -f calico-felix
+	# TODO - update the felix image when https://github.com/projectcalico/calico/pull/1111 is merged
```
1111 was merged

djosborne commented:
Couple remaining things. Then probably LGTM but I'd like to take one final look just to see how they were resolved

tomdee commented:
I've added the gitignore lines and removed the TODO now that PR 1111 is merged.

I'm not going to fix up all the help commands or add more comments - I think this should be merged as is.
